pkg/cloud/openstack/machine/actuator.go

```diff
@@ -94,51 +99,45 @@ func (oc *OpenstackClient) Create(ctx context.Context, cluster *clusterv1.Cluste
 		return nil
 	}
 
-	// get machine startup script
-	var ok bool
-	userData := []byte{}
-	if providerSpec.UserDataSecret != nil {
-		namespace := providerSpec.UserDataSecret.Namespace
-		if namespace == "" {
-			namespace = machine.Namespace
-		}
-
-		if providerSpec.UserDataSecret.Name == "" {
-			return fmt.Errorf("UserDataSecret name must be provided")
-		}
+	userData, err := getUserDataSecret(kubeClient, providerSpec, machine.Namespace)
+	if err != nil {
+		return oc.handleMachineError(machine, apierrors.CreateMachine(
+			"error creating Openstack instance: %v", err))
+	}
 
-		userDataSecret, err := kubeClient.CoreV1().Secrets(namespace).Get(providerSpec.UserDataSecret.Name, metav1.GetOptions{})
+	isMaster := util.IsControlPlaneMachine(machine)
+	var userDataRendered string
+	if len(userData) > 0 {
+		klog.Info("")
+		var err error
+		userDataRendered, err = oc.getMachineScript(isMaster, cluster, machine, string(userData))
 		if err != nil {
-			return err
+			return oc.handleMachineError(machine, apierrors.CreateMachine(
+				"error creating Openstack instance: %v", err))
 		}
-
-		userData, ok = userDataSecret.Data[UserDataKey]
-		if !ok {
-			return fmt.Errorf("Machine's userdata secret %v in namespace %v did not contain key %v", providerSpec.UserDataSecret.Name, namespace, UserDataKey)
+	} else {
+		var distri string
+		if strings.Contains(strings.ToLower(providerSpec.Image), "ubuntu") {
+			distri = "ubuntu"
+		} else if strings.Contains(strings.ToLower(providerSpec.Image), "centos") {
+			distri = "centos"
 		}
-	}
 
-	var userDataRendered string
-	if len(userData) > 0 {
-		if util.IsControlPlaneMachine(machine) {
-			userDataRendered, err = masterStartupScript(cluster, machine, string(userData))
-			if err != nil {
-				return oc.handleMachineError(machine, apierrors.CreateMachine(
-					"error creating Openstack instance: %v", err))
-			}
+		var controlPlaneEndpoint string
+		if isMaster {
+			controlPlaneEndpoint = fmt.Sprintf("%s:443", providerSpec.FloatingIP)
```

Comment:
understand 443 is https... but should make it configurable? or left a TODO here

Reply:
This is hardcoded to 443 in the upstream cluster-api code. There is an issue to make it configurable, but currently we are stuck with the hard coded port